src/acl_auto_configure.cpp

```diff
@@ -493,8 +493,28 @@ static bool read_device_global_mem_defs(
           read_uint32_counters(config_str, curr_pos, dev_global_size, counters);
     }
 
-    acl_device_global_mem_def_t dev_global_def = {dev_global_addr,
-                                                  dev_global_size};
+    // read device global properties
```

Comment:
Sorry I can't comment on the unchanged code, so I will leave my comment here. I am just wondering for the device global address, is there a particular reason that you use uint32_t instead of uint64_t? 

On the compiler side, I have set it to uint64_t, and we also run a test to print the address of device global, e.g: 0x4040000000000000. Therefore, I suppose we should reach an agreement between runtime and compiler, any remark on what specific size of address we should use? Thanks.

Reply:
Good question. I tried to look into the [device global spec](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_device_global.asciidoc) but didn't find any mentioning of the address size.

There is an unmerged change for supporting device global copy kernel though #65, and the device global address is used in `clEnqueueReadGlobalVariableINTEL` and `clEnqueueWriteGlobalVariableINTEL`. At the very end it gets casted to a `void *` whose size is dependent on platform architecture. 

I would say it should be fine for the runtime to adhere to whatever compiler is doing, but we might need to consider how the conversion is handled on a 32-bit host machine. Would you mind explain a little on how the device global address is generated?